### PR TITLE
add docu

### PR DESCRIPTION
**Pull request by sven0311**
add some more docu

### CODE REVIEW

**Review by terlan98 — APPROVED** [review]
The documentation looks good 👍 

I have a non-blocking suggestion: I think it would be useful to add a list of clients along with their repository links and a short description to clarify the apps that are referred to throughout the documentation (`App`, `Themis`, and `Exam Checker`).

I would add something like this right before the 'Modules Overview':

> The package is currently used by the following iOS clients:
> - [Artemis iOS App](https://github.com/ls1intum/artemis-ios) (referred to as the `App`) – Native iOS application for Artemis
> - [Artemis Exam Participation Checker App](https://github.com/ls1intum/ArtemisExamChecker) (referred to as the `Exam Checker`) – an iPad app designed to streamline the process of checking student attendance during on-site exams conducted with the Artemis platform
> - [Themis](https://github.com/ls1intum/Themis)– an iPad app that provides a comprehensive and optimized way for tutors to assess student submissions on the go.

File: README.md [hunk 1]
```diff
@@ -42,6 +42,43 @@ dependencies: [
...
 
-# How to use Modules
+# Module Architecture
+
+![Subsystem decomposition of the Core Module](docu/CORE-MODULE-subsystem.png)
+
+*Figure: Subsystem decomposition of the `Core Module`.*
+
+#### Common Utility
+The Common Utility subsystem provides essential utility functions, encompassing a network state wrapper, logging functionality, and date formatting. Almost all other `Core Module`s extensively utilize this subsystem. Furthermore, all clients, including `App`, `Exam Checker`, and `Themis`, use this subsystem.
```
Comment (nityanandaz) [inline README.md:52]:
```suggestion
The Common Utility subsystem provides essential utility functions, encompassing a network state wrapper, logging functionality, and date formatting.
Almost all other `Core Module`s extensively utilize this subsystem.
Furthermore, all clients, including `App`, `Exam Checker`, and `Themis`, use this subsystem.
```

File: README.md [hunk 2]
```diff
@@ -42,6 +42,43 @@ dependencies: [
...
+
+#### Shared Models
+The Shared Models subsystem houses all the Artemis domain models required by any of the clients. It includes computed properties with design elements for domain objects, such as the exercise icon, varying based on the exercise type. This subsystem is crucial for network communication and its data encryption. It also ensures consistency and efficiency in data handling across different clients. All clients within the system utilize the Shared Models subsystem.
```
Comment (nityanandaz) [inline README.md:55]:
```suggestion
The Shared Models subsystem houses all the Artemis domain models required by any of the clients.
It includes computed properties with design elements for domain objects, such as the exercise icon, varying based on the exercise type.
This subsystem is crucial for network communication and its data encryption.
It also ensures consistency and efficiency in data handling across different clients. All clients within the system utilize the Shared Models subsystem.
```

File: README.md [hunk 3]
```diff
@@ -42,6 +42,43 @@ dependencies: [
...
+
+#### User Session Store
+The User Session Store subsystem stores user-session-related data, encompassing login data and push notification keys. Additionally, it houses configurations about the Artemis instances and maintains a record of the selected instance. It is important to note that this subsystem is exclusively utilized within the `Core Module`, with no direct usage by any client.
```
Comment (nityanandaz) [inline README.md:58]:
```suggestion
The User Session Store subsystem stores user-session-related data, encompassing login data and push notification keys.
Additionally, it houses configurations about the Artemis instances and maintains a record of the selected instance.
It is important to note that this subsystem is exclusively utilized within the `Core Module`, with no direct usage by any client.
```

File: README.md [hunk 4]
```diff
@@ -42,6 +42,43 @@ dependencies: [
...
+
+#### API Client
+The API Client subsystem is a lightweight API client built upon Apple's URLSession. It provides a straightforward method for sending REST requests and supports the content-types 'application/json' and 'multipart/form-data'. This subsystem is responsible for encrypting requests and decryption of responses, ensuring secure data transmission. Additionally, it handles logging and some aspects of error handling, contributing to the system's robustness. The API Client also includes a simple STOMP web socket client, facilitating real-time communication. The API Client subsystem is critical for all other subsystems that necessitate network communication. Its role is to ensure seamless and efficient data exchange within the system, enhancing its overall functionality and performance.
```
Comment (nityanandaz) [inline README.md:61]:
```suggestion
The API Client subsystem is a lightweight API client built upon Apple's URLSession.
It provides a straightforward method for sending REST requests and supports the content-types 'application/json' and 'multipart/form-data'.
This subsystem is responsible for encrypting requests and decryption of responses, ensuring secure data transmission.
Additionally, it handles logging and some aspects of error handling, contributing to the system's robustness.
The API Client also includes a simple STOMP web socket client, facilitating real-time communication.
The API Client subsystem is critical for all other subsystems that necessitate network communication.
Its role is to ensure seamless and efficient data exchange within the system, enhancing its overall functionality and performance.
```

File: README.md [hunk 5]
```diff
@@ -42,6 +42,43 @@ dependencies: [
...
+
+*[STOMP documentation](https://stomp.github.io/stomp-specification-1.2.html)*
+
+#### Shared Services
+The Shared Services subsystem operates with the API Client to deliver service functionality to all other subsystems. This approach ensures that common API calls are not repetitively implemented across different subsystems, promoting code reuse and efficiency. The subsystem comprises an *AccountService*, *CourseService*, and *ExerciseService*, each providing the necessary calls for their respective services. The `App` and `Themis` clients extensively utilize this subsystem.
```
Comment (nityanandaz) [inline README.md:66]:
```suggestion
The Shared Services subsystem operates with the API Client to deliver service functionality to all other subsystems.
This approach ensures that common API calls are not repetitively implemented across different subsystems, promoting code reuse and efficiency.
The subsystem comprises an *AccountService*, *CourseService*, and *ExerciseService*, each providing the necessary calls for their respective services.
The `App` and `Themis` clients extensively utilize this subsystem.
```

File: README.md [hunk 6]
```diff
@@ -42,6 +42,43 @@ dependencies: [
...
+
+#### Design Library
+The Design Library subsystem is a comprehensive collection of UI elements that mirror the aesthetics of the `Webapp`. It encompasses a variety of components, such as buttons, text fields, chips, progress bars, and card modifiers. Additionally, it houses commonly utilized styles, including paddings, colors, and font sizes. It also encapsulates some UIKit components like the 'QLPreviewController' or 'WKWebView', offering SwiftUI versions of these elements. This subsystem is integral to all other subsystems and clients that incorporate UI, enhancing the visual consistency and user experience across the system.
+
+#### Artemis Markdown
+The Artemis Markdown subsystem provides a MarkdownView, a crucial component for rendering the partially custom markdown Artemis utilizes. This subsystem is adept at handling and displaying this custom markdown. Currently, the `App` exclusively employs this module to display lecture descriptions or messages within the instant message feature. This selective use may expand in the future as the system evolves.
```
Comment (nityanandaz) [inline README.md:72]:
```suggestion
The Artemis Markdown subsystem provides a MarkdownView, a crucial component for rendering the partially custom markdown Artemis utilizes. This subsystem is adept at handling and displaying this custom markdown.
Currently, the `App` exclusively employs this module to display lecture descriptions or messages within the instant message feature.
This selective use may expand in the future as the system evolves.
```

File: README.md [hunk 7]
```diff
@@ -42,6 +42,43 @@ dependencies: [
...
+
+#### Login
+The Login subsystem manages the authentication flow of Artemis, providing both the user interface for login and the corresponding service layer. It facilitates institution selection, enabling users to switch between different Artemis instances. All three clients, `App`, `Exam Checker`, and `Themis`, employ this subsystem. Further, the login subsystem utilizes the User Session Store to retain authentication information.
```
Comment (nityanandaz) [inline README.md:75]:
```suggestion
The Login subsystem manages the authentication flow of Artemis, providing both the user interface for login and the corresponding service layer.
It facilitates institution selection, enabling users to switch between different Artemis instances.
All three clients, `App`, `Exam Checker`, and `Themis`, employ this subsystem.
Further, the login subsystem utilizes the User Session Store to retain authentication information.
```

File: README.md [hunk 8]
```diff
@@ -42,6 +42,43 @@ dependencies: [
...
+
+#### Artemis Profile Info
+The Artemis Profile Info subsystem primarily supports the Login subsystem by verifying the compatibility of the chosen Artemis instance with the client. This subsystem is service-oriented and does not incorporate any user interface components.
```
Comment (nityanandaz) [inline README.md:78]:
```suggestion
The Artemis Profile Info subsystem primarily supports the Login subsystem by verifying the compatibility of the chosen Artemis instance with the client.
This subsystem is service-oriented and does not incorporate any user interface components.
```

**Review by nityanandaz — COMMENTED** [review]
I think, shorter lines make tracking changes easier :)

File: README.md [hunk 9]
```diff
@@ -49,36 +57,62 @@ dependencies: [
...
+Further, the login subsystem utilizes the User Session Store to retain authentication information.
 
 #### Artemis Profile Info
-The Artemis Profile Info subsystem primarily supports the Login subsystem by verifying the compatibility of the chosen Artemis instance with the client. This subsystem is service-oriented and does not incorporate any user interface components.
+The Artemis Profile Info subsystem primarily supports the Login subsystem by verifying the compatibility of the chosen Artemis instance with the client.
+This subsystem is service-oriented and does not incorporate any user interface components.
 
 #### Account
-The Account subsystem manages user account operations. It provides the functionality to view account information and log out of the current user. A navigation bar menu, integrated across all clients, facilitates the detail view. 
+The Account subsystem manages user account operations. It provides the functionality to view account information and log out of the current user.
```
Comment (terlan98) [inline README.md:111]:
```suggestion
The Account subsystem manages user account operations. 
It provides the functionality to view account information and log out of the current user.
```

File: README.md [hunk 10]
```diff
@@ -49,36 +57,62 @@ dependencies: [
...
 
 #### Artemis Markdown
-The Artemis Markdown subsystem provides a MarkdownView, a crucial component for rendering the partially custom markdown Artemis utilizes. This subsystem is adept at handling and displaying this custom markdown. Currently, the `App` exclusively employs this module to display lecture descriptions or messages within the instant message feature. This selective use may expand in the future as the system evolves.
+The Artemis Markdown subsystem provides a MarkdownView, a crucial component for rendering the partially custom markdown Artemis utilizes. This subsystem is adept at handling and displaying this custom markdown.
```
Comment (terlan98) [inline README.md:96]:
```suggestion
The Artemis Markdown subsystem provides a MarkdownView, a crucial component for rendering the partially custom markdown Artemis utilizes. 
This subsystem is adept at handling and displaying this custom markdown.
```

**Review by terlan98 — CHANGES_REQUESTED** [review]
I found a few sentences that can be split into new lines (as suggested by Nityananda). The rest looks good.

**Review by terlan98 — APPROVED** [review]
Looks good 👍 Thanks for addressing my comments

File: README.md [hunk 11]
```diff
@@ -42,6 +50,71 @@ dependencies: [
...
+#### Common Utility
+The Common Utility subsystem provides essential utility functions, encompassing a network state wrapper, logging functionality, and date formatting.
+Almost all other `Core Module`s extensively utilize this subsystem.
+Furthermore, all clients, including `App`, `Exam Checker`, and `Themis`, use this subsystem.
+
+#### Shared Models
+The Shared Models subsystem houses all the Artemis domain models required by any of the clients.
+It includes computed properties with design elements for domain objects, such as the exercise icon, varying based on the exercise type.
+This subsystem is crucial for network communication and its data encryption.
+It also ensures consistency and efficiency in data handling across different clients. All clients within the system utilize the Shared Models subsystem.
```
Comment (nityanandaz) [inline README.md:68]:
```suggestion
It also ensures consistency and efficiency in data handling across different clients.
All clients within the system utilize the Shared Models subsystem.
```

**Review by nityanandaz — APPROVED** [review]
Thanks

**Pull request merged**